### PR TITLE
fix: npm link does not respect --no-save

### PR DESCRIPTION
**Pull request by Santoshraj2**
Issue: https://github.com/npm/cli/issues/3619
there was no condition to capture no-save scenario, read that from CLI and included condition to check that which will end up not saving anything if -no-save is passed as arguments.

### CODE REVIEW

File: lib/commands/link.js [hunk 1]
```diff
@@ -108,6 +108,7 @@ class Link extends ArboristWorkspaceCmd {
     // npm link should not save=true by default unless you're
     // using any of --save-dev or other types
     const save =
+      !Boolean(this.npm.config.argv[3] === '--no-save') &&
```
Comment (wraithgar) [inline lib/commands/link.js]: This could be anywhere on argv

Comment (wraithgar) [reply lib/commands/link.js]: This needs to be a proper check against the `save` config value, both for if it has been changed from the default AND if it's been changed to true.

Comment (Santoshraj2) [reply lib/commands/link.js]: Fixed

File: test/lib/commands/link.js [hunk 2]
```diff
@@ -315,14 +315,13 @@ t.test('link pkg already in global space', async t => {
   })
 
   // XXX: how to convert this to a config that gets passed in?
-  npm.config.find = () => 'default'
+  npm.config.find = () => '--save'
```
Comment (wraithgar) [inline test/lib/commands/link.js]: This was an existing test that was testing something in a global context, it's not appropriate to change it to ALSO try to cover this use case.  A net new test will be needed.

Comment (Santoshraj2) [reply test/lib/commands/link.js]: Added new test case.

File: test/lib/commands/link.js [hunk 3]
```diff
@@ -369,6 +369,42 @@ t.test('link pkg already in global space when prefix is a symlink', async t => {
...
+    prefixDir: {
+      'package.json': JSON.stringify({
+        name: 'my-project',
+        version: '1.0.0',
+      }),
+    },
+  })
+
+  // XXX: how to convert this to a config that gets passed in?
+  npm.config.find = () => '--save'
```
Comment (wraithgar) [inline test/lib/commands/link.js]: This is not a valid return value for this function.

File: test/lib/commands/link.js [hunk 4]
```diff
@@ -369,6 +369,42 @@ t.test('link pkg already in global space when prefix is a symlink', async t => {
   t.matchSnapshot(await printLinks(), 'should create a local symlink to global pkg')
 })
 
+t.test('saving link to package file', async t => {
```
Comment (wraithgar) [inline test/lib/commands/link.js]: Where is `save` being set to `false` in this test?

**Comment by wraithgar** [comment]
It looks like we may be hitting coverage by accident here. I would expect a test that sets `save` to false, and has `find` return `cli`, and then assert that it is not saved.

File: test/lib/commands/link.js (hunk 4 above)
Comment (Santoshraj2) [reply test/lib/commands/link.js]: done

File: test/lib/commands/link.js (hunk 3 above)
Comment (Santoshraj2) [reply test/lib/commands/link.js]: removed

File: test/lib/commands/link.js [hunk 5]
```diff
@@ -369,6 +369,40 @@ t.test('link pkg already in global space when prefix is a symlink', async t => {
   t.matchSnapshot(await printLinks(), 'should create a local symlink to global pkg')
 })
 
+t.test('should not saving link to package file', async t => {
```
Comment (wraithgar) [inline test/lib/commands/link.js]:
```suggestion
t.test('should not save link to package file', async t => {
```

**Pull request merged**